File: stacks/nodejs/nestjs-express-reactive/src/services/crypt/crypt.service.ts
import { Injectable } from '@nestjs/common';

const { crypt } = require('utils');
const { v4 } = require('uuid');

@Injectable()
export class CryptService {
  public process() {
    const plain = v4();
    return crypt.encrypt(plain).then((hashed) => crypt.validate(plain, hashed));
  }
}

File: stacks/rust/actix/src/crypt_service.rs
use uuid::Uuid;

use crate::crypt;

pub fn process() {
    let plain: String = Uuid::new_v4().to_hyphenated().to_string();
    let hash = crypt::encrypt(&plain);
    let valid = crypt::validate(&plain, hash);
    assert_eq!(true, valid)
}
